### ISSUE TITLE
support 'Signature' HTTP header?

### ISSUE THREAD

**Issue by snarfed** [ahknight/httpsig#9]
hi! first off, thanks for maintaining and working on this library. we appreciate it!

i'm implementing a protocol ([ActivityPub](https://w3c.github.io/activitypub/)) and using HTTP Signatures via the [`Signature` HTTP header](https://tools.ietf.org/html/draft-cavage-http-signatures-07#section-4), but httpsig currently only supports the `Authorization` header. any chance you could add `Signature` support?

as a proof of concept, i've done this in snarfed/httpsig@6f0ebfe00af39fb75d50c7a29f46da49e8190d70. example usage in snarfed/bridgy-fed@b1230881588b9d990db9fa13f13376b5624c69e6. it's definitely not the way you'd do it for real, but it does work.

thanks in advance!

**Comment by ahknight** [comment]
It's my understanding that the spec requires the use of the Authorization header. Was there an update that changed it?

**Comment by snarfed** [comment]
the [httpsig spec](https://datatracker.ietf.org/doc/draft-cavage-http-signatures/?include_text=1)? i've read and re-read the relevant sections a few times now, and it's hard to follow the language, but i think the key section is 1.1 (emhasis mine):

> This specification provides two mechanisms that can be used by a server to authenticate a client.  The first is the _'Signature' HTTP Authentication Scheme_, which may be used for interactive sessions.
>
> The second is the _Signature HTTP Header_, which is typically used by automated software agents.

_'Signature' HTTP Authentication Scheme_ refers to section 3, the `Authorization` HTTP header. _Signature HTTP Header_ refers to section 4, the `Signature` HTTP header. So if I'm parsing that right, both are supported, neither is always required. my use case here is server to server, which _automated software agents_ seems appropriate for.

specifically, my use case was interacting with Mastodon over ActivityPub, which iirc accepted the `Signature` header but not `Authorization`.

**Comment by snarfed** [comment]
the [httpsig spec](https://datatracker.ietf.org/doc/draft-cavage-http-signatures/?include_text=1)? i've read and re-read the relevant sections a few times now, and it's hard to follow the language, but i think the key section is 1.1 (emhasis mine):

> This specification provides two mechanisms that can be used by a server to authenticate a client.  The first is the _'Signature' HTTP Authentication Scheme_, which may be used for interactive sessions.
>
> The second is the _Signature HTTP Header_, which is typically used by automated software agents.

_'Signature' HTTP Authentication Scheme_ refers to section 3, the `Authorization` HTTP header. _Signature HTTP Header_ refers to section 4, the `Signature` HTTP header. So if I'm parsing that right, both are supported, neither is always required. my use case here is server to server, which _automated software agents_ seems appropriate for.

specifically, my use case was [interacting with Mastodon over ActivityPub](https://github.com/tootsuite/mastodon/issues/4906#issuecomment-328844846), which iirc accepted the `Signature` header but not `Authorization`.

**Comment by ahknight** [comment]
Merged #15.
